lib/semantics/expression.cc

```diff
@@ -136,23 +141,46 @@ common::IfNoLvalue<MaybeExpr, WRAPPED> TypedWrapper(
 class ArgumentAnalyzer {
 public:
   explicit ArgumentAnalyzer(ExpressionAnalyzer &context) : context_{context} {}
-  bool success() const { return success_; }
+  bool fatalErrors() const { return fatalErrors_; }
   ActualArguments &&GetActuals() {
-    CHECK(success_);
+    CHECK(!fatalErrors_);
     return std::move(actuals_);
   }
-  template<typename T> void Analyze(const T &x) {
-    actuals_.emplace_back(context_.Analyze(x));
-    success_ &= actuals_.back().has_value();
+  Expr<SomeType> GetAsExpr(std::size_t i) const {
+    return DEREF(actuals_[i].value().UnwrapExpr());
+  }
+  void Analyze(const common::Indirection<parser::Expr> &x) {
+    Analyze(x.value());
+  }
+  void Analyze(const parser::Expr &x) {
+    actuals_.emplace_back(AnalyzeExpr(x));
+    fatalErrors_ |= !actuals_.back().has_value();
   }
   void Analyze(const parser::ActualArgSpec &, bool isSubroutine);
 
+  bool IsIntrinsicRelational(RelationalOperator);
+  bool IsIntrinsicLogical();
+  bool IsIntrinsicNumeric();
+  bool IsIntrinsicConcat();
+
+  // Find and return a user-defined operator for opr or report an error.
+  // `error` is used if there is no such operator.
```

Comment:
`error` doesn't appear as an argument name in the following prototypes.

Reply:
These get passed as r-value references to `NumericOperation`, etc. so we would need a copy anyway.

Reply:
I've updated the comment.